### ISSUE TITLE
ZeroMQSocketListener and ZeroMQSocketHandler examples in the Logging Cookbook not working

### ISSUE THREAD

**Issue by pablogsal** [python/cpython#75475]
BPO | [31294](https://bugs.python.org/issue31294)
--- | :---
Nosy | @tiran, @pablogsal
PRs | <li>python/cpython#3229</li><li>python/cpython#3430</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = <Date 2017-09-08.22:27:33.661>
created_at = <Date 2017-08-28.18:55:35.156>
labels = ['3.7', 'docs']
title = 'ZeroMQSocketListener and ZeroMQSocketHandler examples in the Logging Cookbook not working'
updated_at = <Date 2017-09-08.22:27:33.658>
user = 'https://github.com/pablogsal'
```

bugs.python.org fields:
```python
activity = <Date 2017-09-08.22:27:33.658>
actor = 'christian.heimes'
assignee = 'docs@python'
closed = True
closed_date = <Date 2017-09-08.22:27:33.661>
closer = 'christian.heimes'
components = ['Documentation']
creation = <Date 2017-08-28.18:55:35.156>
creator = 'pablogsal'
dependencies = []
files = []
hgrepos = []
issue_num = 31294
keywords = ['patch']
message_count = 4.0
messages = ['300960', '301629', '301641', '301745']
nosy_count = 3.0
nosy_names = ['christian.heimes', 'docs@python', 'pablogsal']
pr_nums = ['3229', '3430']
priority = 'normal'
resolution = 'fixed'
stage = 'resolved'
status = 'closed'
superseder = None
type = None
url = 'https://bugs.python.org/issue31294'
versions = ['Python 3.6', 'Python 3.7']
```

</p></details>

**Comment by pablogsal** [comment]
There are two documentation issues regarding the ZeroMQSocketListener and ZeroMQSocketHandler examples in the Logging Cookbook:


The first issue is that in the 'init' method for the 'ZeroMQSocketListener' the base class 'init' is never called and therefore the internal variables are not correctly initialized. This leads to an empty 'handlers' attribute and a undefined 'queue' attribute.

The second issue is that zmq.Socket.send and zmq.Socket.setsockopt only admits bytecode variables and therefore it fails in Python3 when using plain strings.

**Comment by tiran** [comment]
New changeset 586c0502b5eb9a39cabe0bc2707a8ff63114265c by Christian Heimes (Pablo Galindo) in branch 'master':
bpo-31294: Fix ZeroMQSocketListener and ZeroMQSocketHandler examples (bpo-3229)
https://github.com/python/cpython/commit/586c0502b5eb9a39cabe0bc2707a8ff63114265c

**Comment by tiran** [comment]
New changeset 27ce5a1b1931b670da234c30d24bfbbc93fa24d7 by Christian Heimes (Miss Islington (bot)) in branch '3.6':
[3.6] bpo-31294: Fix ZeroMQSocketListener and ZeroMQSocketHandler examples (GH-3229) (bpo-3430)
https://github.com/python/cpython/commit/27ce5a1b1931b670da234c30d24bfbbc93fa24d7

**Comment by tiran** [comment]
Thanks for your contribution. Your PR has been merged into 3.6 and master, which will become Python 3.7 eventually.
